### PR TITLE
Implement navigation module

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
Make the sidebar content come from the `DaffNavigationModule` instead of being hardcoded.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by damienwebdev** [comment]
@lderrickable can you update this to use the new `link-set` component?

**Comment by lderrickable** [comment]
build fails

**Comment by lderrickable** [comment]
This should be ok now @damienwebdev

File: apps/demo/src/app/drivers/in-memory-services/in-memory.service.spec.ts [hunk 1]
```diff
@@ -139,17 +131,20 @@ describe('Driver | In Memory | InMemoryService', () => {
...
     let cartReturn;
     let orderReturn;
+    let navigationReturn;
     let result;
 
     beforeEach(() => {
       productReturn = 'productReturn';
       cartReturn = 'cartReturn';
       orderReturn = 'orderReturn';
+      navigationReturn = 'orderReturn';
```
Comment (damienwebdev) [inline apps/demo/src/app/drivers/in-memory-services/in-memory.service.spec.ts]: `navigationReturn`

Comment (lderrickable) [reply apps/demo/src/app/drivers/in-memory-services/in-memory.service.spec.ts]: fixed

**Pull request merged**